api/app/models.py

```diff
@@ -0,0 +1,115 @@
+# bucketlist/models.py
+
+
+from flask import Flask, g
+from flask.ext.sqlalchemy import SQLAlchemy
+
+from passlib.apps import custom_app_context as pwd_context
+from itsdangerous import (TimedJSONWebSignatureSerializer
+                          as Serializer, BadSignature, SignatureExpired)
+from flask.ext.httpauth import HTTPBasicAuth
+
+auth = HTTPBasicAuth()
+
+# Initialise Flask
+app = Flask(__name__)
+
+db = SQLAlchemy(app)
+
+class User(db.Model):
+    """User model"""
+    id = db.Column(db.Integer, primary_key=True)
+    username = db.Column(db.String(20), unique=True)
+    password = db.Column(db.String(128))
+
+    def hash_password(self, password):
+        """Hash the password to be saved in db"""
+        return pwd_context.encrypt(password)
+
+    def verify_password(self, password):
+        return pwd_context.verify(password, self.password)
+
+    def generate_auth_token(self, expiration = 3600):
+        s = Serializer(app.config['SECRET_KEY'], expires_in = expiration)
+        return s.dumps({ 'id': self.id })
+
+    @staticmethod
+    def verify_auth_token(token):
+        s = Serializer(app.config['SECRET_KEY'])
+        try:
+            data = s.loads(token)
+        except SignatureExpired:
+            return None # valid token, but expired
+        except BadSignature:
+            return None # invalid token
+        user = User.query.get(data['id'])
+        return user
+
+    #  Constructor for User
+    def __init__(self, username, password):
+        self.username = username
+        self.password = self.hash_password(password)
+
+    def __repr__(self):
+        return '<User {0}>'.format(self.username)
+
+
+class Bucketlist(db.Model):
+    """Model for Bucketlist"""
+    id = db.Column(db.Integer, primary_key=True)
+    name = db.Column(db.String(20), unique=True)
+    date_created = db.Column(db.DateTime,
+                             default=db.func.current_timestamp())
+    date_modified = db.Column(db.DateTime,
+                              default=db.func.current_timestamp())
+    created_by = db.Column(db.String(20))
```

Comment:
**Code Quality: Critical**

This should be a foreign key field linked to the user model